test/shortest_path_test.dart

```diff
@@ -11,9 +11,19 @@ Matcher _throwsAssertionError(messageMatcher) =>
     throwsA(const TypeMatcher<AssertionError>()
         .having((ae) => ae.message, 'message', messageMatcher));
 
-int _xKey(X input) => input.value;
+bool _xEquals(X a, X b) {
+  if (a == null) {
```

Comment:
```dart
  if (a == null) return b == null;
  if (b == null) return false;
  return a.value == b.value;
```

Unless we think it's OK to treat a null `value` the same as a null `X` (we do in the hash code).

```dart
bool _xEquals(X a, X b) => a?.value == b?.value;
```